### ISSUE TITLE
wildcard in JSON file

### ISSUE THREAD

**Issue by CharlesCato** [adieyal/sd-dynamic-prompts#619]
I don't know if I do something wrong, but I can't have a wildcard working as soon as it is in json format.
I tried with the exemple given in the doc (and corrected the ' after Van Goth into ") but when I refresh wildcards, nothing appears in the list. No clothing, nor artists, nor anything with the filename (called it main.json).
I wanted to try json to test if the if/else/then was working with wildcards.

{
	"clothing": [
		"T-shirt",
		"Pants",
		"Shoes"
	],
	"artists": {
		"finnish": [
			"Akseli Gallen-Kallela",
			"Eero Järnefelt",
			"Helene Schjerfbeck"
		],
		"dutch": [
			"Piet Mondrian",
    			"Rembrandt van Rijn",
    			"Vincent van Gogh'
		]
	}
}

Thank you for your time !

**Comment by akx** [comment]
Do you see the entries appear in the Wildcards Manager tab?

**Comment by CharlesCato** [comment]
> Do you see the entries appear in the Wildcards Manager tab?

> when I refresh wildcards, nothing appears in the list. No clothing, nor artists, nor anything with the filename (called it main.json).

**Comment by CharlesCato** [comment]
I also check the file on an online json-validator without issues.

**Comment by akx** [comment]
> when I refresh wildcards, nothing appears in the list. No clothing, nor artists, nor anything with the filename (called it main.json).

Then I'd say the file is not found. 

> I've searched in the discussions and it seems nobody ever tried json format or speak about it anyway. Are we even sure it's working or....

I copied the JSON from your post, pasted it into a file,

```
$ pbpaste > extensions/sd-dynamic-prompts/wildcards/issue619.json
```
and the wildcards are correctly populated:
![Screenshot 2023-09-15 at 12 25 45](https://github.com/adieyal/sd-dynamic-prompts/assets/58669/54899d3f-f30b-4a8c-b082-866c0db399e9) ![Screenshot 2023-09-15 at 12 25 40](https://github.com/adieyal/sd-dynamic-prompts/assets/58669/1e767aa9-62d9-47d1-b6db-bfcca28ce191)

**Comment by CharlesCato** [comment]
That's very strange. Don't know why it's not opening json files on my config then. :(

**Comment by CharlesCato** [comment]
Well... I give up json. Not working with my config. There is no point trying more. Not sure what to do with the issue though.
Stil unresolved, but since I don't know anybody who could try with vlado repo+amd, I'm stucked.

**Comment by akx** [comment]
Just to make sure, you are putting the JSON in the extension's wildcard directory? Are there any errors in the console?

**Comment by CharlesCato** [comment]
Yeah... same place as my other wilcards files. No errors concerning DynaPrompts.
Some for dynamic thresholding and script/hook.py.
But nothing I can fix manually. These errors come and go as upgrades come. I could try to updrage again. But it never worked anyway. Even with 0 errors.

**Comment by Autantpourmoi** [comment]
same thing to me , can't see any .json file in that directory.  Did some research on the net to check if there was a special way of using this .json file but didn't find anything

**Comment by CharlesCato** [comment]
> .json or .yaml to .txt
Well they all uses different syntaxes though. If your yaml isn't valid, it won't show.

**Comment by Seikikai** [comment]
Is Someone Currently Working On Fix This Problem?
